zookeeper-server/src/main/java/org/apache/zookeeper/server/Request.java

```diff
@@ -100,6 +100,8 @@ public Request(long sessionId, int xid, int type, TxnHeader hdr, Record txn, lon
 
     public long syncQueueStartTime;
 
+    public long requestThrottleQueueTime;
```

Comment:
I might miss something here, but I don't see where this value is actually get set anywhere in this patch. Without this value being set, I don't think we'll know when a request will be throttled or not ...

Reply:
You've saved my PR again :) Thanks!

Reply:
:) 

Would be good to add a test case that cover the real logic of `shouldThrottleOp` instead of using a mock; if we have such test it will catch issue like this but this is non blocking and latest change lgtm.